codegen-server/python/src/main/kotlin/software/amazon/smithy/rust/codegen/server/python/smithy/generators/PythonServerStructureGenerator.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.server.python.smithy.generators
+
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.StructureShape
+import software.amazon.smithy.model.traits.ErrorTrait
+import software.amazon.smithy.rust.codegen.rustlang.RustWriter
+import software.amazon.smithy.rust.codegen.rustlang.asType
+import software.amazon.smithy.rust.codegen.rustlang.documentShape
+import software.amazon.smithy.rust.codegen.rustlang.render
+import software.amazon.smithy.rust.codegen.rustlang.rustBlock
+import software.amazon.smithy.rust.codegen.rustlang.rustTemplate
+import software.amazon.smithy.rust.codegen.server.python.smithy.PythonServerCargoDependency
+import software.amazon.smithy.rust.codegen.smithy.CodegenContext
+import software.amazon.smithy.rust.codegen.smithy.RuntimeType
+import software.amazon.smithy.rust.codegen.smithy.RustSymbolProvider
+import software.amazon.smithy.rust.codegen.smithy.expectRustMetadata
+import software.amazon.smithy.rust.codegen.smithy.generators.StructureGenerator
+import software.amazon.smithy.rust.codegen.smithy.rustType
+import software.amazon.smithy.rust.codegen.util.hasTrait
+
+/**
+ * To share structures defined in Rust with Python, `pyo3` provides the `PyClass` trait.
+ * This class generates input / output / error structures definitions and implements the
+ * `PyClass` trait.
+ */
+open class PythonServerStructureGenerator(
+    model: Model,
+    private val codegenContext: CodegenContext,
+    private val symbolProvider: RustSymbolProvider,
+    private val writer: RustWriter,
+    private val shape: StructureShape
+) : StructureGenerator(model, symbolProvider, writer, shape) {
+    private val codegenScope =
+        arrayOf(
+            "pyo3" to PythonServerCargoDependency.PyO3.asType(),
+            "SmithyPython" to PythonServerCargoDependency.SmithyHttpServerPython(codegenContext.runtimeConfig).asType()
+        )
+
+    override fun renderStructure() {
```

Comment:
I agree for what it concerns `Enums` (which I fixed). Here we are doing a lot more than just adding the attribute:

* We are adding `#[pyclass]`
* We are adding `#[pyclass]` inheriting from `Exception` in case of errors
* We are adding #[pyo3(get, set)] to attributes

I can workaround the `#[pyclass]` attribute easily, but not so much the `get/set`..

Reply:
I think tacking on attributes is best accomplished through decorators. If the structures we're generating are fundamentally different / additional, then inheritance, `CodegenTarget` or splitting are better approaches.